### ISSUE TITLE
Add activity indicators to taskbar and system tray icons

### ISSUE THREAD

**Issue by amalg** [nextcloud/talk-desktop#391]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are interested into the same feature.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

---

The Win32/64 version of Desktop client does not have any indications on the tray icon for new message counts. Other messaging channels like Facebook Messenger, WhatsApp, Skype, etc. will all show some sort of indicator that something has happened in the application that the user should look at (receipt of a new message).

**Describe the solution you'd like**
- I would like to have the application icon displayed in the taskbar have an indicator as to the number of unread new messages in the client application.
- I would like the system tray icon to display a small contrast colored dot in the upper left or upper right area of the icon to indicate new activity in the client application the user should attend to.

**Describe alternatives you've considered**
Alternatively, instead of a new message counter, the taskbar icon of the application could simply display a colored dot that contrasts with the application icon color to indicate new activity the user should be made aware of.

**Additional context**

Example of Nextcloud Talk with new unread messages compared to other communication clients:

![image](https://github.com/nextcloud/talk-desktop/assets/725551/00b758d9-7d26-4e3f-855a-8d2aea199a8c)

**Comment by git529** [comment]
this is a must have

**Comment by vagner-dias** [comment]
+1

**Comment by djaniol** [comment]
Really appreciate this feature. In general talk is not too bad for usage however without notification like blinking talk icon the whole idea with application for win doesn't have much sense in general in my opinion. 
Some activity indication is very appreciated.

**Comment by Saueee** [comment]
This issue should be priority. Talk has already problems with notification.
I guess people download the dektop version to be notified correctly if something going on.
No point on having the dektop app if it has the same problem.

**Comment by Saueee** [comment]
Well sorry let me rephrase. 
Talk has very few configuration options for notifications.

With problems i mean:
1. 30 sec polling seems too slow. Should be configurable without building source.
2. Browser Tab showing an Asterisk instead of blinking like everything else which is much harder to notice.
3. Incoming calls should have an constant sound/notification until dismissed or accepted (if channel not muted). 
    Currently if you missed the short sound you cant notice at all even if active in nextcloud. Could be configurable too.
4. New group notification default is "mention only" which is too less in most cases. Default setting should be configurable.

**Comment by nickvergessen** [comment]
> 30 sec polling seems too slow. Should be configurable without building source.

https://github.com/nextcloud/notify_push will help and make it instant

> Browser Tab showing an Asterisk instead of blinking like everything else which is much harder to notice.

We can not make it blink, the browsers offer very little options, changing the title will make pinned tabs show a "marker" 
![grafik](https://github.com/nextcloud/talk-desktop/assets/213943/d71c43c1-3a13-4e32-a754-65eb468690de)
And that is all we can do atm as far as I know

> Incoming calls should have an constant sound/notification until dismissed or accepted (if channel not muted).
Currently if you missed the short sound you cant notice at all even if active in nextcloud.

This is still open and we are working towards it

> New group notification default is "mention only" which is too less in most cases. Default setting should be configurable.

![grafik](https://github.com/nextcloud/talk-desktop/assets/213943/10a565a6-171e-48cd-b34c-6655b9653ed3)
It's the first admin option, but yeah missing from user level options atm.

**Comment by arielchai** [comment]
Agreed, some indicator for unread notifications (e.g. red/green dot), would be godsend.
Thanks for the hard work guys :thumbsup:

**Comment by drlight17** [comment]
> Agreed, some indicator for unread notifications (e.g. red/green dot), would be godsend. Thanks for the hard work guys 👍

While waiting this desktop app to contain some core messanger functionality to use in production in the company I did [small Nextcloud Talk wrapper](https://github.com/drlight17/talk-electron) app. It's early alpha but for now it is more suitable for my needs. Unread messages are shown like this:

![image](https://github.com/user-attachments/assets/92cc3b6c-1036-48ee-bbbe-63b9041d9c26)

There are also some behaviour and gui improvements.

**Comment by arielchai** [comment]
> While waiting this desktop app to contain some core messanger functionality to use in production in the company I did [small Nextcloud Talk wrapper](https://github.com/drlight17/talk-electron) app. It's early alpha but for now it is more suitable for my needs. Unread messages are shown like this:
> 
> ![image](https://github.com/user-attachments/assets/92cc3b6c-1036-48ee-bbbe-63b9041d9c26)
> 
> There are also some behaviour and gui improvements.

Awesome, Notifications work great!

I did have an issue, which I posted on your github, thank you so much for sharing mate 👍

**Comment by manswiss** [comment]
The desktop app has now been officially released and we would like to use it in our company. But unfortunately it is not yet possible without this feature.

**Comment by ER-EPR** [comment]
@drlight17 Will the [wrapper](https://github.com/drlight17/talk-electron) be updated to the 1.0 version? as [talk-desktop](https://github.com/nextcloud/talk-desktop)

**Comment by drlight17** [comment]
> @drlight17 Will the [wrapper](https://github.com/drlight17/talk-electron) be updated to the 1.0 version? as [talk-desktop](https://github.com/nextcloud/talk-desktop) or are they not related. Can I use it in my small studio?

They are not related. Wrapper development is independent.
I had a plan to use something from wrapper code to implement in official app, but it takes too much time to understand how everything works here. Also official app is too heavy imho as it needs all spreed source code to work and to be built. Wrapper app don't. But both will not operate if there is no connection with server. Official can show cached dialogs anyway.
And yes, of course you can use wrapper app as you want. New versions are possible to support Nextcloud compatibility and maybe some bug fixes. I use it in some orgs as there is also domain sso feature available.

**Comment by himself1985** [comment]
I really need this feature to make Nextcloud Talk Desktop truly useful in our daily workflow. Right now, the lack of reliable notifications makes it tough to rely on the app for important communications. Implementing this feature would be a game-changer for me, allowing me to use Nextcloud Talk Desktop as my primary communication tool without worrying about missing messages. This isn’t just a nice-to-have—it’s a must-have for staying productive.

**Comment by fraenki** [comment]
Now this is the most sought-after feature in Talk Desktop. 🥇 And my teammates don't stop complaining about it 😆

**Comment by manswiss** [comment]
We still use Skype internally. However, this will be discontinued in two months and has not been working properly recently. We therefore want to replace it very soon.
However, this function is absolutely essential for this, otherwise we will unfortunately have to look for another solution.
For this reason, I would be very, very happy if this could be implemented as soon as possible.
I am aware that you are doing a great deal and investing a lot of time in this product. I am very grateful for that.

**Comment by drlight17** [comment]
To anyone who is interested. As time passes I've managed to make an update of [my version of Talk Desktop wrapper client](https://github.com/drlight17/talk-electron/releases/latest) to match actual version of Nexcloud Talk. Activity indicators are still operatable however =)
There are some new features like using user OS password wallet to save credentials in. Check [CHANGES.md](https://raw.githubusercontent.com/drlight17/talk-electron/main/CHANGES.md)

**Comment by nxmndr** [comment]
Hopefully this feedback helps finishing it. It doesn't work on my colleague's laptop. The badge appears and then disappears about a second later (without changing window focus or doing anything), which defeats the point.

Specs :
- MacOS 14.5 Sonoma (build 23F79)
- latest Talk 1.2.5
- OS notifications permissions enabled
- Talk's notifications settings for all messages
- reboots regularly

While it works for me (MacOS 11.5.2 Big Sur), though not always.
